front/lib/api/assistant/pubsub.ts

```diff
@@ -165,61 +182,68 @@ export async function retryAgentMessageWithPubSub(
     conversation: ConversationType;
     message: AgentMessageType;
   }
-): Promise<AgentMessageType> {
-  const promise: Promise<AgentMessageType> = new Promise((resolve, reject) => {
-    void (async () => {
-      const redis = await redisClient();
-      let didResolve = false;
-      try {
-        for await (const event of retryAgentMessage(auth, {
-          conversation,
-          message,
-        })) {
-          switch (event.type) {
-            case "agent_message_new": {
-              const pubsubChannel = getConversationChannelId(conversation.sId);
-              await redis.xAdd(pubsubChannel, "*", {
-                payload: JSON.stringify(event),
-              });
-              await redis.expire(pubsubChannel, 60 * 10);
-              didResolve = true;
-              resolve(event.message);
-              break;
-            }
-            case "retrieval_params":
-            case "agent_error":
-            case "agent_action_success":
-            case "generation_tokens":
-            case "agent_generation_success":
-            case "agent_message_success": {
-              const pubsubChannel = getMessageChannelId(event.messageId);
-              await redis.xAdd(pubsubChannel, "*", {
-                payload: JSON.stringify(event),
-              });
-              await redis.expire(pubsubChannel, 60 * 10);
-              break;
+): Promise<Result<AgentMessageType, APIErrorWithStatusCode>> {
```

Comment:
Since we always resolve now, is that const promise with a function call needed? I think directly returning would work no?

Reply:
Not in an async loop